src/OpenSSL/SSL.py

```diff
@@ -161,6 +164,53 @@ class SysCallError(Error):
     pass
 
 
+_CFFI_VERSION_RE = re.compile(r"^(\d+)\.(\d+).*$")
+
+
+def _make_buffer_checker(cffi_version_string):
+    """
+    Returns a callable that converts a sendable buffer object into something
+    that can be sent by OpenSSL using CFFI. Has different supported types
+    depending on the CFFI version.
+
+    :param cffi_version_string: The version of CFFI in use, from
+        ffi.__version__.
+    """
+    cffi_version = tuple(
+        int(component) for component in
+        _CFFI_VERSION_RE.match(cffi_version_string).group(1, 2)
```

Comment:
@hynek The main reason is that you can conceivably have a version string like `1.8dev4`, which will not behave nicely under `.split('.')`. It's possible I'm being too defensive here, but I was trying to avoid having everything explode in situations I didn't entirely forsee.

@reaperhulk That seems reasonable to me, if it's ok with @hynek I'll happily do that instead.